src/extensions/default/bramble/lib/MouseManager.js

```diff
@@ -0,0 +1,230 @@
+    var LiveDevProtocol = brackets.getModule("LiveDevelopment/MultiBrowserImpl/protocol/LiveDevProtocol");
+    var CommandManager = brackets.getModule("command/CommandManager");
+    var Commands = brackets.getModule("command/Commands");
+    var FileSystem = brackets.getModule("filesystem/FileSystem");
+
+    var MouseManagerRemote = require("text!lib/MouseManagerRemote.js");
+    var IframeBrowser = require("lib/iframe-browser");
+
+    var _isInspectorEnabled = false;
+    var _prevLineMarker;
+    var _prevElemHighlightId;
+
+    // Override the preview document's cursor property
+    function setIframeCursor(value) {
+        var iframe;
+        var doc;
+
+        iframe = IframeBrowser.getBrowserIframe();
+        if(!iframe) {
+            return;
+        }
+
+        doc = iframe.contentDocument || iframe.contentWindow.document;
+        if(!doc) {
+            return;
+        }
+
+        doc.body.style.cursor = value;
+    }
+
+    // Depending on the state of the inspector, apply the appropriate cursor
+    function ensurePreviewCursor() {
+        if(_isInspectorEnabled) {
+            setIframeCursor("pointer");
+        } else {
+            setIframeCursor("auto");
+        }
+    }
+
+    // Make sure the editor is showing the current LiveDoc HTML for inspector
+    function ensureLiveDoc() {
+        // Preview path
+        var currentDoc = LiveDevMultiBrowser._getCurrentLiveDoc();
+        var liveDocPath = currentDoc.doc.file.fullPath;
+        // Editor path
+        var currentEditor = EditorManager.getActiveEditor();
+        var editorPath = currentEditor.document.file.fullPath;
+
+        // If they already match, nothing to do
+        if(liveDocPath === editorPath) {
+            return;
+        }
+
+        // Otherwise, switch editor to LiveDoc's HTML file
+        FileSystem.resolve(liveDocPath, function(err, file) {
+            if(err) {
+                console.log("[Bramble Error] unable to open path in editor", liveDocPath, err);
+                return;
+            }
+            CommandManager.execute(Commands.CMD_ADD_TO_WORKINGSET_AND_OPEN, { fullPath: file.fullPath });
+        });
+    }
+
+    function getCodeMirror() {
+        var editor = EditorManager.getActiveEditor();
+        return editor._codeMirror;
+    }
+
+    // Clear text marks in the editor, including ones we've added with inspector
+    function clearMarks() {
+        var codeMirror = getCodeMirror();
+
+        // Clear any text we've marked
+        clearLineHighlight();
+
+        // Get rid of multiple selections in the editor
+        var doc = codeMirror.getDoc();
+        var pos = doc.getCursor("start");
+        codeMirror.setCursor(pos);
+    }
+
+    // When the user re-focuses the editor, clear any marks we've set with inspector
+    function clearMarksOnFocus() {
+        var codeMirror = getCodeMirror();
+
+        function onFocus() {
```

Comment:
Not quite.  I want to be able to remove the listener as soon as it fires, so I wanted a named function.  I use the line below because the instance of CodeMirror might have changed (user switches editor to another file?).  Maybe it's safe to reuse the `codeMirror` instance from 91?  What do you think?